app-libs/stf/src/stf_sgx.rs

```diff
@@ -15,32 +15,34 @@
 
 */
 
-use crate::{
-	helpers::{
-		account_data, account_nonce, ensure_root, get_account_info, get_storage_value,
-		increment_nonce, root, validate_nonce,
-	},
-	AccountData, AccountId, Getter, Index, PublicGetter, ShardIdentifier, State, StateTypeDiff,
-	Stf, StfError, StfResult, TrustedCall, TrustedCallSigned, TrustedGetter,
-};
 use codec::Encode;
-use itp_settings::node::{TEEREX_MODULE, UNSHIELD};
-use itp_storage::storage_value_key;
-use itp_types::OpaqueCall;
 use log_sgx::*;
 use sgx_externalities::SgxExternalitiesTrait;
-use sgx_runtime::{BlockNumber as L1BlockNumer, Runtime};
+use sgx_runtime::Runtime;
 use sgx_tstd as std;
 use sp_io::hashing::blake2_256;
-use sp_runtime::MultiAddress;
+use sp_runtime::{traits::Header, MultiAddress};
 use std::{prelude::v1::*, vec};
 use support::traits::UnfilteredDispatchable;
 
-#[cfg(feature = "test")]
-use crate::test_genesis::test_genesis_setup;
+use itp_settings::node::{TEEREX_MODULE, UNSHIELD};
+use itp_storage::storage_value_key;
+use itp_types::OpaqueCall;
 use its_primitives::types::{BlockHash, BlockNumber as SidechainBlockNumber, Timestamp};
 use its_state::SidechainSystemExt;
 
+#[cfg(feature = "test")]
+use crate::test_genesis::test_genesis_setup;
+use crate::{
+	helpers::{
+		account_data, account_nonce, ensure_root, get_account_info, get_parentchain_blockhash,
+		get_parentchain_number, get_parentchain_parenthash, get_storage_value, increment_nonce,
+		root, validate_nonce,
+	},
+	AccountData, AccountId, Getter, Index, ParentchainHeader, PublicGetter, ShardIdentifier, State,
+	StateTypeDiff, Stf, StfError, StfResult, TrustedCall, TrustedCallSigned, TrustedGetter,
+};
+
```

Comment:
Why do we separate these `use`? Is it because they are the local `crate::` imports? Let the cargo formatter handle it (with the current settings, it will always put them at the top) and don't try to add you own ordering by separating them. The problem being, that you own ordering will always be different from other developers. The ordering cargo fmt chooses will always be consistent.

Reply:
The first one is for tests.